File: WarehouseManagementWeb.Application/Dto/Output/ResourceReceipt/ResourceReceiptListOutput.cs
namespace WarehouseManagementWeb.Application.Dto.Output.ResourceReceipt
{
    /// <summary>
    /// Класс выходной модели ресурсов поступлений.
    /// </summary>
    public class ResourceReceiptListOutput
    {
        /// <summary>
        /// Id документа поступления.
        /// </summary>
        public int DocumentReceiptId { get; set; }

        /// <summary>
        /// Номер документа поступления.
        /// </summary>
        public string? DocumentReceiptNumberCode { get; set; }

        /// <summary>
        /// Дата документа поступления.
        /// </summary>
        public DateTime DocumentReceiptDate { get; set; }

        /// <summary>
        /// Наименование клиента документа поступения.
        /// </summary>
        public string? DocumentClientName { get; set; }

        /// <summary>
        ///  Список входящих ресурсов.
        /// </summary>
        public IEnumerable<ResourceReceiptItemOutput>? Items { get; set; }
    }
}

File: warehouse-management-web-angular/src/app/warehouse/document-receipt/models/output/resource-receipt-list-output.ts
import { ResourceReceiptItemOutput } from './resource-receipt-item-output';

/**
 * Класс выходной модели ресурсов поступлений.
 */
export class ResourceReceiptListOutput {
  /**
   * Id документа поступления.
   */
  documentReceiptId: number = 0;

  /**
   * Номер документа поступления.
   */
  documentReceiptNumberCode: string = '';

  /**
   * Дата документа поступления.
   */
  documentReceiptDate: Date = new Date();

  /**
   * Id клиента документа поступления.
   */
  documentReceiptClientId: number = 0;

  /**
   * Наименование клиента документа поступения.
   */
  documentReceiptClientName: string = '';

  /**
   * Список входящих ресурсов.
   */
  items: ResourceReceiptItemOutput[]  = [];
}
